src/movetype.cpp

```diff
@@ -421,9 +421,7 @@ movetype::terrain_info::terrain_info(const config & cfg, const parameters & para
  */
 std::unique_ptr<movetype::terrain_costs> movetype::read_terrain_costs(const config & cfg)
 {
-	// todoc++14: use std::make_unique
-	std::unique_ptr<terrain_costs> ret;
-	ret.reset(new terrain_info(cfg, mvj_params_, nullptr));
+	auto ret = static_cast<std::unique_ptr<terrain_costs>> (std::make_unique<terrain_info> (cfg, movetype::mvj_params_, nullptr));
```

Comment:
I think you should just be able to do
```cpp
return std::make_unique<terrain_info>(cfg, movetype::mvj_params_, nullptr);
```

Reply:
I remember something being a unhappy with that when I tried it on another project, but can't remember which compiler (maybe a linter) complained about it.

I'll switch to that syntax next time I rebase this (I'm intending to leave this PR open so that others can comment on the flies to flying change).